docs/developer/first_delivery.md

```diff
@@ -0,0 +1,241 @@
+
+```typescript
+import { filesChangedSince, goal, GoalInvocation, PushListenerInvocation, pushTest } from "@atomist/sdm";
+```
+
+The `pushListenerInvocation` contains all the information about a project and a push. Within our filtering function, we can call the `filesChangedSince` method and provide it with the necessary arguments:
+
+```typescript
+async (pushListenerInvocation: PushListenerInvocation) => {
+  const changedFiles = await filesChangedSince(pushListenerInvocation.project, pushListenerInvocation.push);
+  return changedFiles.includes("README.md");
+});
+```
+
+For completeness, we may want to check for case-insensitivity when looking for `README.md`, but for this tutorial it should be enough.
+
+Our complete `pushTest` method should look like this:
+
+```typescript
+import { filesChangedSince, goal, GoalInvocation, PushListenerInvocation, pushTest } from "@atomist/sdm";
+
+const modifiesReadme = pushTest(
+  "modifiesReadme",
+  async (pushListenerInvocation: PushListenerInvocation) => {
+    const changedFiles = await filesChangedSince(pushListenerInvocation.project, pushListenerInvocation.push);
+    return changedFiles.includes("README.md");
+  });
+```
+
+## Attaching to your SDM
+
+We've set up a goal and defined when we want it to be invoked; it's time to associate it with our SDM!
+
+The _lib/machine/machine.ts_ file is used to control the behavior of the SDM. Here, we can define which events we want to listen to and which goals to run for those events.
+
+Let's start by importing the two functions that we defined earlier:
+
+```typescript
+import {
+    messageGoal, modifiesReadme,
+} from "./message";
+```
+
+Next, within the `machine` method, enter the following line:
+
+```typescript
+sdm.withPushRules(
+    whenPushSatisfies(modifiesReadme).setGoals(messageGoal));
+```
+
+The method names provide clarity on the configuration:
+
+* we configure the SDM to listen to push events (`withPushRules`)
+* we test that the push in question passes our previous filter (`whenPushSatisfies`)
+* if the push does pass the filter, we set the goals we want the SDM to achieve (`setGoals`)
+
+That's all there is to it!
+
+## Associating the SDM with your repository
+
+Now that our SDM is set up the behave just as we want it to, it's time to test everything and make sure that our code is correct. Before doing that, though, it's important to go over the differences between [Local Mode](/developer/local/) and [Team Mode](/developer/team/). Local Mode runs in the privacy of your own machine. The SDM doesn't communicate with the online Atomist service, and it behaves just like a local instance of the SDM would. Crucially, since we're operating locally, push events are simulated as coming from _commits_.
+
+To associate the local `atomist-seeds/express-es6-rest-api` repository with the SDM running Local Mode, you'll need to navigate to the folder and enter:
```

Comment:
Because I was told to fork then `atomist clone`, this isn't in `atomist-seeds` for me. Prolly just shorten to `express-es6-rest-api`.